### PR TITLE
Fix manifest sorting in chart renderer

### PR DESCRIPTION
**Pull request by timebertt**
**What this PR does / why we need it**:
With this PR, the Chart Renderer splits up rendered chart template files into individual manifests before sorting them by Kind.
This is done to allow proper sorting of the manifests (for example: VPA CRD should be applied before VPA CRs). Previously the chart renderer only sorted rendered files as a whole and not individual manifests. This can cause issues when applying the rendered manifests in case there are multiple manifests of different kinds in a single file.

**Which issue(s) this PR fixes**:
Ref #1939

**Special notes for your reviewer**:
I will open a follow-up PR for removing the early exit on error in the chart applier.

**Release note**:

```improvement developer
The Chart Renderer now splits up rendered template files into individual manifests to ensure proper sorting of the manifests.
```

### CODE REVIEW

**Review by rfranzke — DISMISSED** [review]
/lgtm

File: pkg/chartrenderer/default.go [hunk 1]
```diff
@@ -141,14 +142,21 @@ func (r *chartRenderer) renderResources(ch *chartapi.Chart, values chartutil.Val
...
 		if strings.HasSuffix(k, notesFileSuffix) || strings.HasPrefix(path.Base(k), "_") {
 			delete(files, k)
 		}
 	}
 
-	manifests := manifest.SplitManifests(files)
+	manifestsMap := map[string]string{}
+	for origKey, file := range files {
+		for key, m := range releaseutil.SplitManifests(file) {
+			manifestsMap[origKey+"#"+key] = m
```
Comment (ialidzhikov) [inline pkg/chartrenderer/default.go]: I thinks the controllerinstallation control is using this pkg and `#` in the manifest key breaks it:

```
time="2020-02-19T10:12:49+02:00" level=info msg="Error syncing ControllerInstallation provider-gcp-5pq58: Secret \"provider-gcp-5pq58\" is invalid: [data[provider-gcp_templates_vpa.yaml#manifest-0]: Invalid value: \"provider-gcp_templates_vpa.yaml#manifest-0\": a valid config key must consist of alphanumeric characters, '-', '_' or '.' (e.g. 'key.name',  or 'KEY_NAME',  or 'key-name', regex used for validation is '[-._a-zA-Z0-9]+'),
```

**Review by ialidzhikov — CHANGES_REQUESTED** [review]
Generally it is a nice catch.

File: pkg/chartrenderer/default.go (hunk 1 above)
Comment (timebertt) [reply pkg/chartrenderer/default.go]: Thanks, nice catch.
I was not sure, which separator to pick...
WDYT: should we use `/` (will be replaced with `_` in ctrlinst controller) or should we use `_` directly?

Comment (ialidzhikov) [reply pkg/chartrenderer/default.go]: Both of them looks good to me.

**Comment by timebertt** [comment]
Fixed the map keys to use `_` as a separator.
ControllerInstallations work again now. The ctrlinst secret keys will look like this: `gardener-extension-provider-mock_templates_configmap.yaml_manifest-0`.

PTAL :)

**Review by ialidzhikov — APPROVED** [review]
/lgtm

**Pull request merged**